http/rest-client-reactive-vanilla/src/main/java/io/quarkus/ts/http/restclient/reactive/RestCallerService.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.ts.http.restclient.reactive;
+
+import java.net.URI;
+
+import jakarta.annotation.PostConstruct;
+
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+
+import io.quarkus.runtime.Startup;
+
+/**
+ * This class is used to reproduce issue https://github.com/quarkusio/quarkus/issues/31024
+ * Also requires {@link UselessRestApi} and {@link VersionHeaderFilter} for this
+ */
+@Startup
+public class RestCallerService {
+
+    @PostConstruct
+    void init() {
+        callRest();
+    }
+
+    void callRest() {
```

Comment:
The name _callRest_ is not really precise as there is no call. I don't insist on changing it, just saying :-)

Reply:
I renamed it, and further simplified this class.